src/dags/wior.py

```diff
@@ -0,0 +1,248 @@
+import operator
+import requests
+from requests.auth import HTTPBasicAuth
+import json
+import logging
+
+from airflow import DAG
+from airflow.models import Variable
+from airflow.operators.bash_operator import BashOperator
+from airflow.operators.python_operator import PythonOperator
+from airflow.operators.postgres_operator import PostgresOperator
+
+from common.db import DatabaseEngine
+from environs import Env
+
+from ogr2ogr_operator import Ogr2OgrOperator
+from provenance_rename_operator import ProvenanceRenameOperator
+from pgcomparator_cdc_operator import PgComparatorCDCOperator
+from sqlalchemy_create_object_operator import SqlAlchemyCreateObjectOperator
+
+
+from common import (
+    default_args,
+    slack_webhook_token,
+    DATAPUNT_ENVIRONMENT,
+    SHARED_DIR,
+    MessageOperator,
+)
+
+from postgres_check_operator import (
+    PostgresMultiCheckOperator,
+    COUNT_CHECK,
+    GEO_CHECK,
+)
+
+from sql.wior import (
+    DROP_COLS,
+    SQL_DROP_TMP_TABLE,
+    SQL_GEOM_CONVERT,
+    SQL_ADD_PK,
+    SQL_SET_DATE_DATA_TYPES,
+)
+
+dag_id = "wior"
+variables = Variable.get(dag_id, deserialize_json=True)
+data_endpoint = variables["data_endpoints"]["wfs"]
+tmp_dir = f"{SHARED_DIR}/{dag_id}"
+data_file = f"{tmp_dir}/{dag_id}.geojson"
+env = Env()
+db_conn = DatabaseEngine()
+password = env("AIRFLOW_CONN_WIOR_PASSWD")
+user = env("AIRFLOW_CONN_WIOR_USER")
+base_url = env("AIRFLOW_CONN_WIOR_BASE_URL")
+total_checks = []
+count_checks = []
+geo_checks = []
+
+
+# define logger for output to console
+logger = logging.getLogger(__name__)
+
+
+# needed to put quotes on elements in geotypes for SQL_CHECK_GEO
+def quote(instr: str) -> str:
+    return f"'{instr}'"
+
+
+class DataSourceError(Exception):
+    pass
+
+
+# data connection
+def get_data() -> None:
+    """calling the data endpoint"""
+    # get data
+    data_url = f"{base_url}/{data_endpoint}"
+    data_request = requests.get(data_url, auth=HTTPBasicAuth(user, password))
+    # store data
+    if data_request.status_code == 200:
+        try:
+            data = data_request.json()
+        except json.decoder.JSONDecodeError:
+            logger.error(f"Failed to convert request output to json for url {data_url}")
+            raise
```

Comment:
Generally it is a good idea to use exception chaining;

```python
except json.decoder.JSONDecodeError as jde:
    ...
    raise from jde
```

Reply:
Explicieter :+1:  aangepast